yargs.js

```diff
@@ -661,6 +671,11 @@ function Yargs (processArgs, cwd, parentRequire) {
       if (~handlerKeys.indexOf(cmd) && cmd !== completionCommand) {
         setPlaceholderKeys(argv)
         return command.runCommand(cmd, self, parsed)
+      } else {
+        if (recommendCommands) {
+          const similiarCommand = didYouMean(cmd, handlerKeys)
```

Comment:
Perhaps we should defer the `require('didyoumean')` until it's actually used here. That would avoid loading an unused dependency in cases where the CLI doesn't even use commands.